### ISSUE TITLE
Supermoon

### ISSUE THREAD

**Issue by tailorshivam** [shred/commons-suncalc#27]
how to find next Super moon date?

**Comment by shred** [comment]
suncalc does not support Super Moon calculation at the moment. You can use `MoonPosition.getDistance()` to get the moon distance for a certain date, but you would have to search for the date with the closest distance yourself.

I think Super Moon should be supported by suncalc, though. I will have a look into that, but it may take a while. I will keep this issue open as a reminder.

**Comment by isomeme** [comment]
Since there's [no consistent definition](https://en.wikipedia.org/wiki/Supermoon#Definitions) of a supermoon, you'd need to provide a time or distance based metric, and perhaps make it configurable by callers. I think "instant of full moon is within (duration) of instant of perigee" would probably be the easiest to code. But most of the definitions in use seem to be based on distance from mean perigee at the instant of full moon.

**Comment by shred** [comment]
Oh yes, I forgot that the moon phase needs to be considered as well. :sweat_smile:

But I think using `MoonPhase` is a good approach for a start. In commit 4d193102, the geocentric distance to the moon is now a part of the result. Having the distance, one can find out if the next full moon is a super moon according to one's favorite definition:

```
MoonPhase mp = MoonPhase.compute().phase(Phase.FULL_MOON).execute();
if (mp.getDistance() < 360000.0) { // less than 360,000 km according to timeanddate.com
  System.out.println("Next full moon will be a super moon!");
}
```

There is a problem though. _suncalc_ uses a simplified formula for calculating the moon's distance, which does not consider any perturbations.

**This means that currently, the accuracy is not sufficient for a super or micro moon calculation.**

I need to find a formula with a better approximation to the moon's true distance. I will appreciate any help, of course! :wink:

**Comment by shred** [comment]
Good news: I found a better formula for moon distance calculation (see 92738a41). :smiley:

`MoonPhase` now also got two new methods `.isSuperMoon()` and `.isMicroMoon()` (see 29170bd8), which use the timeanddate.com definition of supermoon and micromoon. In the documentation, the _Moon Phase_ example has been extended to use the new methods.

The changes will be released in suncalc v3.4. I will also backport them to suncalc v2.11.

@tailorshivam Is your problem resolved by these changes?

**Comment by tailorshivam** [comment]
@shred Thank you so much for your support
